rabbitsmpp/consumer.go

```diff
@@ -93,30 +114,47 @@ func (c *consumer) Consume() (<-chan Job, <-chan error, error) {
 		}()
 		defer close(jobChan)
 		defer close(errChan)
-
 		for {
-			select {
-			case d := <-msgs:
-				j := Job{}
-				err := json.Unmarshal(d.Body, &j)
-				if err != nil {
-					// TODO: Figure out what to do with this failed job
-					log.Printf("failed to unmarshal PDU: %v", err)
-					errChan <- err
-					return
-				}
-				j.delivery = &d
-				jobChan <- j
-			case <-c.ctx.Done():
+			err = c.consume(dlvChan, closeChan, jobChan)
+			// if consume returns without an error, means that it was terminated
+			// properly, otherwise something went wrong and it needs to restart
+			if err == nil {
 				log.Printf("EOF consuming for: %s", c.ID())
 				return
 			}
+			errChan <- err
```

Comment:
I am still unsure if this errorChan is necessary for the overall functionality of the Consumer, but it may be useful for callers that want to know when a consumer is failing (even though it restarts itself).